### DISCUSSION TITLE
Can I update the src content of the audio component?

### DISCUSSION THREAD

**Discussion by KimigaiiWuyi** [zauberzeug/nicegui#2444] [Ideas / Feature Requests]
### Question

Can I update the src content of the audio component?

**Comment by falkoschindler** [comment]
Hi @KimigaiiWuyi,

There's currently no official API to change the source, but you can access the private `_props` dictionary like this:

```py
a = ui.audio('https://cdn.pixabay.com/download/audio/2022/02/22/audio_d1718ab41b.mp3')

def change_source() -> None:
    a._props['src'] = 'https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3'
    a.update()

ui.button('Change src', on_click=change_source)
```

Anyway, I'll convert this discussion into a feature request. It shouldn't be too hard to add a getter and setter for such attributes.

**Reply by KimigaiiWuyi** [reply]
In fact, I've tried this method before coming over here to ask about it, and this method doesn't actually work
After `a.update()`, I get a grayed out, defunct, audio component

**Reply by falkoschindler** [reply]
Strange. This exact code works for me in Chrome and Safari. 🤔

**Reply by KimigaiiWuyi** [reply]
I reconfirmed my own code and I noticed the difference，
I tried your code and it does work, but replace the network audio with local audio and it doesn't work

**Reply by falkoschindler** [reply]
Oh, you're right. For local files we need special treatment. As a workaround you can explicitly serve the media files and use their routes as sources:
https://nicegui.io/documentation/section_pages_routing#add_directory_of_media_files_

**Reply by KimigaiiWuyi** [reply]
Yes I can do that, it alleviates the problem temporarily, so will the problem eventually be fixed? Or is it more difficult to fix issuse？

**Reply by falkoschindler** [reply]
It is not too hard and we will implement this feature. I just can't promise it for the next release or two. 🙂

**Reply by KimigaiiWuyi** [reply]
Thank you so much for everything, this has helped me so much, looking forward to the next few releases! 🥰

**Reply by falkoschindler** [reply]
It was easier than expected: PR #2449 simply lets `ui.audio` and `ui.video` inherit from `SourceElement`, which already does everything we need automatically.

**Reply by KimigaiiWuyi** [reply]
💖Wow, glad to hear the good news, that's great! I'll be switching from pypi to git to complete the fix! Thanks for doing this together, it's very helpful to me!

**Reply by falkoschindler** [reply]
NiceGUI 1.4.13 has just been released. ✔️
